pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -152,23 +155,40 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 var NilClusterPatcher = clusterPatcher{}
 
 type clusterPatcher struct {
-	efw  *model.EnvoyFilterWrapper
-	pctx networking.EnvoyFilter_PatchContext
+	efw            *model.EnvoyFilterWrapper
+	pctx           networking.EnvoyFilter_PatchContext
+	patchesApplied int
 }
 
 func (p clusterPatcher) conditionallyAppend(l []*cluster.Cluster, hosts []host.Name, clusters ...*cluster.Cluster) []*cluster.Cluster {
+	applied := false
+	var pc *cluster.Cluster
 	for _, c := range clusters {
 		if envoyfilter.ShouldKeepCluster(p.pctx, p.efw, c, hosts) {
-			l = append(l, envoyfilter.ApplyClusterMerge(p.pctx, p.efw, c, hosts))
+			pc, applied = envoyfilter.ApplyClusterMerge(p.pctx, p.efw, c, hosts)
```

Comment:
I am a bit confused on logic here, so `applied` will be based on if the last cluster is patched successfully or not?

Reply:
Not it is not. The cluster patcher holds the applied patch count and it knows how many patches it needs to apply - so it will increment that metric for all the skipped patches. See `incrementFilterMetrics` function on patcher. This is because the cluster patcher will try to run through all patches for every cluster - So we can not say if it is not applied for one cluster it is not applied at all. At the end of all clusters we increment the stats.

Reply:
Do you intent to put `if applied { p.patchesApplied++ }` inside the `for c : clusters { .. }` loop? Currently it is outside the loop and that does not make sense to me..

Reply:
Yeah. That is a great catch. Fixed now. PTAL.